External/devspace.yaml

```diff
@@ -0,0 +1,135 @@
+version: v2beta1
+name: edge-external-app-example
+
+vars:
+  DEVSPACE_AWS_PROFILE:
+    source: env
+    default: PlaceholderProfile-000000000000
+  DEVSPACE_AWS_ECR_HOST:
+    source: env
+    default: 000000000000.dkr.ecr.us-east-1.amazonaws.com
+  DEVSPACE_FLAGS: "-n ${DEVSPACE_NAME}"
+  EDM_APP_DEPENDENCIES: "flask setuptools gunicorn"
+
+localRegistry:
+  enabled: false
+
+pipelines:
+  build:
+    flags:
+    - name: rebuild-zbundle
+    run: |-
+      build_images --all
+  dev:
+    flags:
+    - name: rebuild-zbundle
+    run: |-
+      build_images --all
+      create_deployments --all
+      start_dev --all
+  purge:
+    run: |-
+      stop_dev --all
+      purge_deployments --all
+
+hooks:
+  - name: "pre-deploy-hook"
+    command: |-
+      $!(
+        if [ ${DEVSPACE_CONTEXT} != "docker-desktop" ] && [ ${DEVSPACE_CONTEXT} != "minikube" ]; then
+          echo 'aws sts get-caller-identity --profile ${DEVSPACE_AWS_PROFILE} >/dev/null 2>&1 || aws sso login'
+        else
+          echo ''
+        fi
+      )
+    events: ["before:deploy"]
+  - name: "pre-image-build-hook"
+    command: |-
+      $!(
+        if ! [ -f ./src/app_environment.zbundle ] || [ $(get_flag "rebuild-zbundle") == "true" ]; then
+          echo 'edm bundle generate -i --version 3.8 --platform rh7-x86_64 -m 2.0 -f ./src/app_environment.zbundle ${EDM_APP_DEPENDENCIES}'
+        else
+          echo ''
+        fi
+        if [ ${DEVSPACE_CONTEXT} != "docker-desktop" ] && [ ${DEVSPACE_CONTEXT} != "minikube" ]; then
+          echo 'aws sts get-caller-identity --profile ${DEVSPACE_AWS_PROFILE} >/dev/null 2>&1 || aws sso login'
+          echo 'aws ecr get-login-password --profile ${DEVSPACE_AWS_PROFILE} | docker login --username AWS --password-stdin ${DEVSPACE_AWS_ECR_HOST}'
+        else
+          echo ''
+        fi
+      )
+    events: ["before:build"]
```

Comment:
While currently only relevant for remote deployments, the current example config contains a lot of AWS ECR specific logic.

This allows for simple, unified authentication to EKS _and_ ECR via our usual AWS SSO setup, but I'm wondering if this should be vendor-neutral instead (i.e. generic container registry credentials and a K8s pull secret).